runtime/src/main/java/org/corfudb/protocols/wireprotocol/NettyCorfuMessageDecoder.java

```diff
@@ -2,36 +2,30 @@
 
 import io.netty.buffer.ByteBuf;
 import io.netty.buffer.Unpooled;
+import io.netty.channel.ChannelHandler.Sharable;
 import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandler;
 import io.netty.handler.codec.ByteToMessageDecoder;
 
+import io.netty.handler.codec.MessageToMessageDecoder;
 import java.util.List;
 
+import javax.annotation.Nonnull;
 import lombok.extern.slf4j.Slf4j;
 
 
 /**
  * Created by mwei on 10/1/15.
  */
 @Slf4j
-public class NettyCorfuMessageDecoder extends ByteToMessageDecoder {
+@Sharable
```

Comment:
I believe It is not actually enough just to annotate it as 'sharable' (this is only an informational annotation). This would require CorfuServer and NettyClientRouter to add the same instance to the pipeline. Maybe this can explain why we don't actually see improvements in the performance runs.

In the [Documentation of the annotation](https://netty.io/4.0/api/io/netty/channel/ChannelHandler.Sharable.html) you can find this description:

> This annotation is provided for documentation purpose, just like the JCIP annotations.

For more details you can read the "Shared and Exclusive Channel Handlers" of this [blog](http://seeallhearall.blogspot.com/2012/06/netty-tutorial-part-15-on-channel.html)

Reply:
@annym Thanks! I made the change to keep the same encoder/decoder. I wouldn't expect a really significant performance increase until we start using multiple channels.

Reply:
Great! You are right we would see improvements only when using multiple channels.